Header header
int64[] ts
int32   aux_encoder
